hugegraph-llm/src/hugegraph_llm/utils/gradio_demo.py

```diff
@@ -0,0 +1,254 @@
+# under the License.
+
+
+import json
+import os
+
+import gradio as gr
+import uvicorn
+from fastapi import FastAPI
+
+from hugegraph_llm.llms.init_llm import LLMs
+from hugegraph_llm.operators.graph_rag_task import GraphRAG
+from hugegraph_llm.operators.kg_construction_task import KgBuilder
+from hugegraph_llm.utils.config import Config
+from hugegraph_llm.utils.constants import Constants
+from pyhugegraph.client import PyHugeClient
+
+
+def init_hg_test_data():
+    client = get_hg_client()
+    schema = client.schema()
+    schema.propertyKey("name").asText().ifNotExist().create()
+    schema.propertyKey("birthDate").asText().ifNotExist().create()
+    schema.vertexLabel("Person").properties(
+        "name", "birthDate"
+    ).useCustomizeStringId().ifNotExist().create()
+    schema.vertexLabel("Movie").properties("name").useCustomizeStringId().ifNotExist().create()
+    schema.edgeLabel("ActedIn").sourceLabel("Person").targetLabel("Movie").ifNotExist().create()
+
+    schema.indexLabel("PersonByName").onV("Person").by("name").secondary().ifNotExist().create()
+    schema.indexLabel("MovieByName").onV("Movie").by("name").secondary().ifNotExist().create()
+
+    graph = client.graph()
+    graph.addVertex("Person", {"name": "Al Pacino", "birthDate": "1940-04-25"}, id="Al Pacino")
+    graph.addVertex(
+        "Person",
+        {"name": "Robert De Niro", "birthDate": "1943-08-17"},
+        id="Robert De Niro",
+    )
+    graph.addVertex("Movie", {"name": "The Godfather"}, id="The Godfather")
+    graph.addVertex("Movie", {"name": "The Godfather Part II"}, id="The Godfather Part II")
+    graph.addVertex(
+        "Movie",
+        {"name": "The Godfather Coda The Death of Michael Corleone"},
+        id="The Godfather Coda The Death of Michael Corleone",
+    )
+
+    graph.addEdge("ActedIn", "Al Pacino", "The Godfather", {})
+    graph.addEdge("ActedIn", "Al Pacino", "The Godfather Part II", {})
+    graph.addEdge("ActedIn", "Al Pacino", "The Godfather Coda The Death of Michael Corleone", {})
+    graph.addEdge("ActedIn", "Robert De Niro", "The Godfather Part II", {})
+    schema.getSchema()
+    graph.close()
+    return {
+        "vertex": ["Person", "Movie"],
+        "edge": ["ActedIn"],
+        "property": ["name", "birthDate"],
+        "index": ["PersonByName", "MovieByName"],
+    }
+
+
+def graph_rag(text):
+    res = (
+        GraphRAG()
+        .extract_keyword(text=text)
+        .query_graph_for_rag()
+        .synthesize_answer()
+        .run(verbose=True)
+    )
+    return res
+
+
+def build_kg(text, schema, disambiguate_word_sense, commit_to_hugegraph):
+    builder = KgBuilder(LLMs().get_llm())
+    # 判断 schema 是不是json结构
+    if schema:
+        try:
+            schema = json.loads(schema.strip())
+            builder.import_schema(from_user_defined=schema)
+        except json.JSONDecodeError as e:
+            print(e)
+            builder.import_schema(from_hugegraph=schema)
```

Comment:
When we want to import schema from huge graph, we input the graph name as `schema`.
However, `schema` is `str` can be `json.loads` without exception.

Reply:
Importing from hugegraph only needs to specify graph_name, which is not a json structure. Of course, this design is not elegant, and we can find ways to optimize it later.
- from_user_defined: {"xx": "xx"}
- from_hugegraph: "tmp_graph"